app/models/concerns/scorecards/lockable.rb

```diff
@@ -7,17 +7,36 @@ module Scorecards::Lockable
     validate :locked_scorecard, on: :update
 
     def lock_access!
-      self.locked_at = Time.now.utc
+      self.completed_at = Time.now.utc
+      self.progress = "completed"
+
+      save(validate: false)
```

Comment:
@hengsokly your method name convention is lock_aaccess! with the exclamation mark, it means that in the method body should raise an error but at the end your call save(validate: false) where save() return true/false, it should calls save!() instead, please kindly check it.